spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,644 @@
+#
+# Compute maximum spinal cord compression using AP diameter or other morphometrics.
+#
+# ---------------------------------------------------------------------------------------
+# Copyright (c) 2015 Polytechnique Montreal <www.neuro.polymtl.ca>
+# Author: Sandrine Bédard, Jan Valosek, Julien Cohen-Adad
+#
+# About the license: see the file LICENSE.TXT
+#########################################################################################
+# TODO: maybe create an API or move some functions
+import sys
+import os
+import numpy as np
+import csv
+import logging
+from typing import Sequence
+import pandas as pd
+from spinalcordtoolbox.utils import SCTArgumentParser, Metavar, init_sct, set_loglevel
+from spinalcordtoolbox.utils.fs import get_absolute_path, check_file_exist, extract_fname, printv
+from spinalcordtoolbox.image import Image
+from spinalcordtoolbox.centerline.core import get_centerline, ParamCenterline
+from spinalcordtoolbox.types import Centerline
+from spinalcordtoolbox import __data_dir__
+from spinalcordtoolbox.scripts import sct_process_segmentation
+
+
+logger = logging.getLogger(__name__)
+
+
+NEAR_ZERO_THRESHOLD = 1e-6
+
+
+# PARSER
+# ==========================================================================================
+def get_parser():
+    parser = SCTArgumentParser(
+        description='Compute normalized morphometric measures to assess spinal cord compression.\n'
+                    '\n'
+                    'Metrics are normalized using the non-compressed levels above and below the compression site using '
+                    'the following equation:\n'
+                    '\n'
+                    '    ratio = (1 - mi/((ma+mb)/2))\n'
+                    '\n'
+                    'Where mi: metric at the compression level, ma: metric above the compression level, mb: metric '
+                    'below the compression level.\n'
+                    '\n'
+                    'Additionally, if the "-normalize" flag is used, metrics are normalized using a database built '
+                    'from healthy control subjects. This database uses the PAM50 template as an anatomical reference '
+                    'system.\n'
+                    '\n'
+                    'Reference: Miyanji F, Furlan JC, Aarabi B, Arnold PM, Fehlings MG. Acute cervical traumatic '
+                    'spinal cord injury: MR imaging findings correlated with neurologic outcome--prospective study '
+                    'with 100 consecutive patients. Radiology 2007;243(3):820-827.'
+    )
+
+    mandatoryArguments = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatoryArguments.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help='NIfTI file of spinal cord segmentation to compute morphometrics. If no normalization is wanted (if -i-PAM50 is not specified),'
+             ' metric ratio will take average along the centerline.'
```

Comment:
Maybe, since there is now no input .csv file, the help could be simplified:

```suggestion
        help='Spinal cord segmentation mask to compute morphometrics from. Example: t2_seg.nii.gz'
         'Note: If no normalization is wanted (i.e., if the "-i-PAM50" flag is not specified),'
             ' metric ratio will take average along the segmentation centerline.'
```